### ISSUE TITLE
Search box results in app crash

### ISSUE THREAD

**Issue by joejoe135791** [Drumber/Kitsune#69]
### Problem description

Whenever I use the search box, click on a result and attempt to back out, the app crashes

### Steps to reproduce

Go to search
Search anything and click a result 
Use the back button to back out of that result 

### App version

2.0.2

### Android version

14

### Device information

OnePlus 9 pro

### Attachments and Logs

https://github.com/user-attachments/assets/e8dcbca4-b820-4989-b3b5-9319bc5ad9a3

### Acknowledgements

- [X] I have searched the open and closed [issues](https://github.com/Drumber/Kitsune/issues?q=is%3Aissue) and this is **NOT** a duplicate.
- [X] I'm using the latest version of the app.
- [X] I have provided all required information.

**Comment by joejoe135791** [comment]
I also just figured out how to export the logs if this helps

[Kitsune_2024-11-12_23-54-11.txt](https://github.com/user-attachments/files/17728527/Kitsune_2024-11-12_23-54-11.txt)

**Comment by Drumber** [comment]
Thanks, the logs helped a lot.

Unfortunately, I couldn't reproduce the error on an emulator running Android 14. So, I'm not sure if the issue is just related to specific devices.

Based on the stack trace from the logs your shared, I might have an idea what could cause the crash. Can you check if the error is fixed running this debug build: https://github.com/Drumber/Kitsune/actions/runs/11822426352/artifacts/2183328420
(Don't be confused, the debug build has the same name and logo as the release app. You may uninstall it later.)

**Comment by joejoe135791** [comment]
I am still encountering the same issue with the debug build you sent above. If anything, it's even faster at crashing 

[Kitsune_2024-11-13_09-40-38.txt](https://github.com/user-attachments/files/17736982/Kitsune_2024-11-13_09-40-38.txt)

**Comment by joejoe135791** [comment]
looking at the debug log lines 141 and 142 are showing missing files, could that be the reason for the crash?

**Comment by Drumber** [comment]
> looking at the debug log lines 141 and 142 are showing missing files, could that be the reason for the crash?

Thats okay, just some system warning during the install.

The crash is caused by these lines (something related to the fragment not being attached to an activity during onDestroyView)
```
11-13 09:40:08.047 30722 30722 E AndroidRuntime: java.lang.IllegalStateException: Fragment's view can't be accessed. Fragment isn't added
11-13 09:40:08.047 30722 30722 E AndroidRuntime: 	at by.kirich1409.viewbindingdelegate.LifecycleViewBindingProperty.getValue(ViewBindingProperty.kt:87)
11-13 09:40:08.047 30722 30722 E AndroidRuntime: 	at by.kirich1409.viewbindingdelegate.FragmentViewBindingProperty.getValue(FragmentViewBindings.kt:63)
11-13 09:40:08.047 30722 30722 E AndroidRuntime: 	at by.kirich1409.viewbindingdelegate.FragmentViewBindingProperty.getValue(FragmentViewBindings.kt:52)
11-13 09:40:08.047 30722 30722 E AndroidRuntime: 	at io.github.drumber.kitsune.ui.main.MainFragment.getBinding(MainFragment.kt:30)
11-13 09:40:08.047 30722 30722 E AndroidRuntime: 	at io.github.drumber.kitsune.ui.main.MainFragment.onDestroyView(MainFragment.kt:128)
```

Could you try this build and share the logs if it crashes: https://github.com/Drumber/Kitsune/actions/runs/11823001992/artifacts/2183510313

**Comment by joejoe135791** [comment]
It's working for now (other than the occasional leaks pop up). I've been messing with it for a little bit and it's not crashing. Would you like the log?

**Comment by Drumber** [comment]
That's great news! Would be helpfull if you could share the logs (a last time ^^). I'm just curious what's going on here. Might be a bug related to the fragment lifecycle breaking the ViewBindingDelegrate library I'm using.

**Comment by joejoe135791** [comment]
Here's the newest log for you.
[Kitsune_2024-11-13_10-22-55.txt](https://github.com/user-attachments/files/17737476/Kitsune_2024-11-13_10-22-55.txt)

Leaks was also finding other leaks but I assume that's just because it's a debug build and not related to my first issue

**Comment by joejoe135791** [comment]
Thanks so much for the help.

I also just noticed, this was issue number #69. Nice lol

Anyways, have a good one

**Comment by joejoe135791** [comment]
Just updated any while it's not a major issue. Every time I launch the app it's telling br up "upgrade" to version 2.0.2

**Comment by Drumber** [comment]
Yeah well, I need to improve the update checker. It's just comparing the current version with the latest release in GitHub. And since 2.0.3 is a pre-release (and not the test stable release) it's telling you there is an update.

You may want to disable the update checker in the settings until the next stable release. Sry for that...

**Comment by joejoe135791** [comment]
The update fixed it from happening every time however I'm still getting a few crashes 
[Kitsune_2024-11-13_11-41-45.txt](https://github.com/user-attachments/files/17738449/Kitsune_2024-11-13_11-41-45.txt)

**Comment by Drumber** [comment]
Thanks for providing the logs. Seems like this might be a bigger problem. I will look into it when I've more time.

**Comment by Drumber** [comment]
@joejoe135791 did you have any problems with v2.0.?

**Comment by joejoe135791** [comment]
No, I did not encounter this issue on version 2.0.1

**Comment by Drumber** [comment]
Just out of curiosity, did you enable [predictive back animations](https://www.androidpolice.com/turn-on-predictive-back-navigation-android-14/) in the android developer options?

Besides updating the compile SDK of the app to 35 (Android 15) and opting-in to predictive-back-navigation, there weren't any big changes that could cause this bug.

**Comment by Drumber** [comment]
Okay, I've done some bigger code changes and I think the root cause of the crash should be fixed now.

Unfortuanate I still could not reproduce the crash on any device/emulator I have at hand. So, it would be great if you could test this build again and report back if you find any crashes/bugs: https://github.com/Drumber/Kitsune/actions/runs/11842451101/artifacts/2189041831

**Comment by Drumber** [comment]
Looks like the crash is now caused by an edge case where some delayed/async code is accessing a view while the fragment (search screen) got destroyed.

I'm still not sure what is causing these strange view lifecycle problems on your device. I have tried to force the occurrence of the crash in some test scenarios, but no luck so far...

Once again, it would be great if you could test this build and let me know if you are still experiencing crashes: https://github.com/Drumber/Kitsune/actions/runs/11859089583/artifacts/2193742475

**Comment by joejoe135791** [comment]
My previous message I accidentally opened prerelease 2.0.3 instead of the debug build. After messing with the debug build for around 10 minutes, I haven't encountered any major crashes (I was going so fast I accidentally edited the app though). I will update with more info if I crash. Until then, here's the log

[Kitsune_2024-11-15_08-20-45.txt](https://github.com/user-attachments/files/17778218/Kitsune_2024-11-15_08-20-45.txt)

Everything I tried that crashed the other versions, 
* Exiting an anime that's currently loading 
* Exiting and opening another show in quick succession (several times)

**Comment by Drumber** [comment]
Hi @joejoe135791, is the build stable so far, no further crashes?

**Comment by joejoe135791** [comment]
Yes, I have experienced any crashes with the experimental build

**Comment by Drumber** [comment]
Thanks again for your help. I've published a new release [v2.0.4](https://github.com/Drumber/Kitsune/releases/tag/v2.0.4) which includes the fix.

If you encounter any other problems, feel free to post it here (if related) or open a new issue.
